### ISSUE TITLE
Multiline regexp - wrong line reported

### ISSUE THREAD

**Issue by Maxim-Mazurok** [gmullerb/eslint-plugin-regex#11]
I'm using this:
```
'regex/invalid': [
	'error',
	[
		{
			regex: '[`\'"][\\s\\n]*\\{[\\s\\n]*"',
			message: 'No inline JSON is allowed, use JSON.stringify() instead',
			replacement: {
				function: 'return text', // make it unfixable
			},
		},
	],
],
```
to catch these:
```
			body: `{
	"TotalItems":5,
	"TotalPages":2,
	"Items":[
		{
			"UserPk":"101",
			"Email":"u1@mail.box",
			"Name":"user1"
		}
	]
}`,
```

Basically, I'm trying to prevent people from writing inline JSON, and use `JSON.stringify()` instead.

If I run it for this example/repro file:
```
const one = `{
	"TotalItems":5,
}`;
const two = `{
	"TotalItems":5,
}`;
const three = `{
	"TotalItems":5,
}`;
```

I'll get:
```
  1:13  error  No inline JSON is allowed, use JSON.stringify() instead  regex/invalid
  5:13  error  No inline JSON is allowed, use JSON.stringify() instead  regex/invalid
  6:15  error  No inline JSON is allowed, use JSON.stringify() instead  regex/invalid
```

But I expect lines 1, 4, and 7 to be reported. Not 1, 5, and 6.

I was able to fix this by keeping track of the full source file and not using `nextLine` to calculate position.
```
function checkRegex(..., originalSource) {
    originalSource = originalSource || source;
    if (sourceDetail.patternIndex !== -1) {
      if (foundDetail.nextLine !== -1) {
        ...
      }
      else {
        report({
          loc: { start: foundStart(originalSource, 1, sourceDetail.patternIndex + rootChar) },
          ...
        })
        checkRegex(..., originalSource)
      }
    }
  }
}
```

**Comment by gmullerb** [comment]
**Thanks! I really appreciate Your contribution** :1st_place_medal: 
I will check this (issue & PR) with detail next week, so for August 15 we should have this solve.

**Comment by Maxim-Mazurok** [comment]
Hello, @gmullerb just a reminder that I'm looking forward to your review, thanks!
